### PR TITLE
DNS commands

### PR DESCRIPTION
**Pull request by johnmichaelwelsh**
Implementing the dns commands so maestro-shell can be used to set
domain name servers.

### CODE REVIEW

File: unixHttpRoutes.go [hunk 1]
```diff
@@ -83,6 +83,10 @@ func AddProcessRoutes(router *httprouter.Router) {
 	router.GET("/net/events", handleSubscribeNetworkEvents)
 	router.GET("/net/events/:subscription", handleGetLatestNetworkEvents)
 
+	router.GET("/dns", handleGetDNS)
```
Comment (costanic) [inline unixHttpRoutes.go:86]: would "/net/dns" be a more consistent path?

File: networking/manager.go [hunk 2]
```diff
@@ -412,6 +412,125 @@ func (this *networkManagerInstance) StorageClosed(instance storage.MaestroDBStor
...
+	returns
+	error - any errors
+*/
+func (this *networkManagerInstance) DeleteDNS(dns string) error {
+
+	//delete the nameserver in the config
+	this.networkConfig.Nameservers = removeElement(this.networkConfig.Nameservers, dns)
+
+	//make the config active
+	instance.submitConfig(instance.networkConfig)
```
Comment (costanic) [inline networking/manager.go:523]: I see similar code in AddDNS().  Should this be combined?

File: unixHttpRoutes.go [hunk 3]
```diff
@@ -276,6 +280,86 @@ func handleDeleteLogFilter(w http.ResponseWriter, r *http.Request, _ httprouter.
...
+func handlePostDNS(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	inst := networking.GetInstance()
+
+	config := new(maestroSpecs.NetworkConfigPayload)
+
+	if inst != nil {
+		body, err := ioutil.ReadAll(r.Body)
+		if err == nil {
+			json.Unmarshal(body, &config)
+			dns := config.Nameservers[0]
```
Comment (costanic) [inline unixHttpRoutes.go:312]: should this be unrolled to AddDNS() on each Nameserver in the array?

File: unixHttpRoutes.go [hunk 4]
```diff
@@ -276,6 +280,86 @@ func handleDeleteLogFilter(w http.ResponseWriter, r *http.Request, _ httprouter.
...
+func handleDeleteDNS(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	inst := networking.GetInstance()
+
+	config := new(maestroSpecs.NetworkConfigPayload)
+
+	if inst != nil {
+		body, err := ioutil.ReadAll(r.Body)
+		if err == nil {
+			json.Unmarshal(body, &config)
+			dns := config.Nameservers[0]
```
Comment (costanic) [inline unixHttpRoutes.go:342]: same comment as above.  do we want to process the whole array?

File: unixHttpRoutes.go (hunk 1 above)
Comment (johnmichaelwelsh) [reply unixHttpRoutes.go:86]: done

File: networking/manager.go (hunk 2 above)
Comment (johnmichaelwelsh) [reply networking/manager.go]: done.

File: unixHttpRoutes.go (hunk 3 above)
Comment (johnmichaelwelsh) [reply unixHttpRoutes.go]: done

File: unixHttpRoutes.go (hunk 4 above)
Comment (johnmichaelwelsh) [reply unixHttpRoutes.go]: done.

File: unixHttpRoutes.go (hunk 1 above)
Comment (costanic) [inline unixHttpRoutes.go:86]: I forget...did we decide to move this under /net/dns?

**Comment by costanic** [comment]
It seems CI is failing for an unrelated issue:

> [Pipeline] sh
> + vagrant up jenkins_build --provision
> Bringing machine 'jenkins_build' up with 'virtualbox' provider...
> An action 'up' was attempted on the machine 'jenkins_build',
> but another process is already executing an action on the machine.
> Vagrant locks each machine for access by only one process at a time.
> Please wait until the other Vagrant process finishes modifying this
> machine, then try again.
> 
> If you believe this message is in error, please check the process
> listing for any "ruby" or "vagrant" processes and kill them. Then
> try again.

File: unixHttpRoutes.go (hunk 1 above)
Comment (mray190) [reply unixHttpRoutes.go:86]: I think we should for consistency purposes

Comment (johnmichaelwelsh) [reply unixHttpRoutes.go]: done

**Pull request merged**